src/protocols/bdx/BdxTransferServer.cpp

```diff
@@ -40,6 +31,16 @@ CHIP_ERROR BDXTransferServer::ListenForSendInit(System::Layer * systemLayer, Mes
     return mExchangeMgr->RegisterUnsolicitedMessageHandlerForType(MessageType::SendInit, this);
 }
 
+CHIP_ERROR BDXTransferServer::OnUnsolicitedMessageReceived(const PayloadHeader & payloadHeader,
+                                                           Messaging::ExchangeDelegate *& newDelegate)
+{
+    auto * logTransfer = mPoolDelegate.Allocate(mDelegate, mSystemLayer);
```

Comment:
Where's the OnExchangeCreationFailed implementation that deallocates this to avoid leaks?

And if exchange creation succeeds, where is the delegate deallocated?

Reply:
| Where's the OnExchangeCreationFailed implementation that deallocates this to avoid leaks?
Added.

| And if exchange creation succeeds, where is the delegate deallocated?
The deallocation happens when the transfer is terminated right now. I will investigate a bit more to see when it leaks.

Reply:
I move the deallocation in 'OnExchangeClosing'. So now there is both a deallocation in 'OnExchangeCreationFailed' and one in 'OnExchangeClosing'.